s11/array_meth3.rb

```diff
@@ -0,0 +1,76 @@
+# .any?
+p [1, 3, 2, 8].any? {|n| n.even? }
+puts
+
+# .all
+p [1, 3, 5, 7].all? {|n| n.odd? }
+p [1, 2, 5, 7].all? {|n| n.odd? }
+puts
+
+# .find && .detect
+w = %w[air water earth fire]
+p w
+p w.select {|wd| wd.length<5}
+p w.find   {|wd| wd.length<5}
+p w.detect {|wd| wd.length<5}
+p w.reverse.find {|wd| wd.length<5}
+puts
+
+# .uniq
+l = %w[a b a c d c b a]
+p l
+p l.uniq
+puts
+
+# .compact
+p [1, nil, 2, 3]
+p [1, nil, 2, 3].compact
+puts
+
+# .inject && .reduce
+# .reduce
+n = [10, 20, 30]
+aux = n.reduce(0) do |prev,curr|
```

Comment:
Can you explain how `reduce` works?

Reply:
The Reduce method crosses the Array, but also passes by parameter an initial value that after each cycle will take the returned value of the Block. At the end of the iterations, the method returns the final value of the initial parameter.